File: java-sample/src/main/java/io/snabble/testapp/ProjectPaymentOptionsActivity.java
package io.snabble.testapp;

import androidx.fragment.app.Fragment;
import io.snabble.sdk.ui.payment.ProjectPaymentOptionsFragment;

public class ProjectPaymentOptionsActivity extends BaseActivity {
    @Override
    public Fragment onCreateFragment() {
        ProjectPaymentOptionsFragment fragment = new ProjectPaymentOptionsFragment();
        fragment.setArguments(getIntent().getBundleExtra("args"));
        return fragment;
    }
}

File: ui/src/main/java/io/snabble/sdk/ui/payment/ProjectPaymentOptionsActivity.kt
package io.snabble.sdk.ui.payment

import androidx.fragment.app.Fragment
import io.snabble.sdk.ui.BaseFragmentActivity

class ProjectPaymentOptionsActivity : BaseFragmentActivity() {
    companion object {
        const val ARG_BRAND = ProjectPaymentOptionsView.ARG_BRAND
    }

    override fun onCreateFragment(): Fragment {
        val fragment = ProjectPaymentOptionsFragment()
        fragment.arguments = intent.extras
        return fragment
    }
}